Refactored SideBar and improved its styling on mobile

Otju opened the pull request:
Changes:
- SideBar no longer shown on mobile
- SideBar contents now integrated into the NavBar on mobile. 
This probably won't work too well if the SideBar has a loads of content under one header.

Otju committed:
Refactored SideBar, improved styling on mobile

PurkkaKoodari commented on src/components/NavPages.tsx:
```diff
@@ -0,0 +1,94 @@
+  )
+
+  // use browser location to figure out which nav item to show in the sidebar
+  const { pathname } = useLocation()
+  const pathParts = pathname
+    .split("/")
+    .slice(1) // remove empty part before /
+    .filter((part, index) => index !== 0 || part !== locale) // remove locale name as first part
+  const rootItem = nav?.items?.find((item) =>
+    item.path.startsWith(`/${pathParts[0]}`)
+  )
```
This stuff is unnecessary if we know the root node already. Refactor this stuff back to `SideBar` and pass `rootItem` to `NavPages` instead.

PurkkaKoodari commented on src/components/SideBar.tsx:
```diff
@@ -1,94 +1,17 @@
-          <Item key={item.path} item={item} />
-        ))}
-      </nav>
+      <div className="hidden md:inline">
```
```suggestion
      <div className="hidden md:block">
```
(even though there's no difference inside `flex`, it's semantically a block)

PurkkaKoodari commented on src/components/NavBar.tsx:
```diff
@@ -56,7 +59,7 @@ const NavBar: React.FC = () => {
         <div
           className={classNames(
-            "fixed top-0 left-0 h-full bg-gray-darkest transition-[max-width] ease-in-out duration-500",
+            "fixed top-0 left-0 h-full bg-gray-darkest transition-all ease-in-out duration-500 mt-12",
```
Is there a reason this doesn't work with always setting `w-full` and then just toggling `max-width`?

PurkkaKoodari reviewed (commented):
Seems to work, but I think the nav menu shouldn't use the `SideBar` logic.

I've also made some fixes to the layout, committing them in a bit.

PurkkaKoodari committed:
Fixes to side menu styling
- Use .pt-12` on parent instead of .mt-12 to avoid clipping on short
  screens
- Set width on inner menu element to prevent menu items from being
  "squeezed" when opening/closing
- Hide X overflow in menu to prevent scrollbar flashing

Otju committed:
Refactored root node stuff back to SideBar. Added rootItem prop.

PurkkaKoodari commented on src/components/NavBar.tsx:
```diff
@@ -68,18 +76,17 @@ const NavBar: React.FC = () => {
-                {pathname.includes(item.path) && <NavPages />}
-              </>
+                {pathname.includes(item.path) && <NavPages rootItem={item} />}
+              </div>
```
Use a [keyed `Fragment`](https://reactjs.org/docs/fragments.html#keyed-fragments) instead of `div`.

PurkkaKoodari reviewed (approved):
LGTM, there's a couple small nitpicks which I'll fix before merging.

PurkkaKoodari committed:
Fix menu link widths, remove unnecessary div
